youtube_transcript_api/_cli.py

```diff
@@ -26,19 +25,24 @@ def run(self):
         transcripts = []
         exceptions = []
 
+        Formatter = formats.get_formatter(parsed_args.format)
+
         for video_id in parsed_args.video_ids:
             try:
-                transcripts.append(self._fetch_transcript(parsed_args, proxies, cookies, video_id))
+                transcript = self._fetch_transcript(
+                    parsed_args, proxies, cookies, video_id)
+                transcripts.append(Formatter.format(transcript))
             except Exception as exception:
                 exceptions.append(exception)
 
-        return '\n\n'.join(
+        return ''.join(
             [str(exception) for exception in exceptions]
-            + ([json.dumps(transcripts) if parsed_args.json else pprint.pformat(transcripts)] if transcripts else [])
+            + ([Formatter.combine(transcripts)] if transcripts else [])
```

Comment:
I felt the same way when I wrote it. 🙈 It felt like a hacky attempt to make the CLI and API play nice between 1 video and 2 or more videos and because JSON strings were working a little different than regular strings on different payloads, especially if you want to be able to `json.loads`.

Reply:
yeah, I can definitely understand how you got there 😄